### ISSUE TITLE
Pre-built SDK Engine does not include .setreg files from gems

### ISSUE THREAD

**Issue by LB-TomaszSerwanski** [o3de/o3de#4164]
Original title: Newly created Blast Materials fail to open in the Asset Editor and crash the Editor when attempting to save changes

**Describe the bug**
Blast Materials created in the Asset Editor cannot be open in it. When attempting to do so, the Asset Editor window remains blank and the following errors are displayed in the Console:
```
[Error] (AssetDatabase) - No handler was registered for this asset (id={25E43245-6DF7-529B-AC4E-53B2C49F3E86}:0, type={00000000-0000-0000-0000-000000000000})!
[Error] (AssetManager) - BlockUntilLoadComplete must be called after an asset has been queued for load. Asset ({00000000-0000-0000-0000-000000000000}:0) is not queued for load
```
Additionally, when closing the Asset Editor after attempting to load such Blast Material, the _Save Changes_ dialog appears. If the User selects _Yes_, the Editor crashes.

Please refer to the attached video and log files for more details.

**To Reproduce**
1. Open the Asset Editor.
2. Select _File_ -> _New_ -> _Blast Material_.
3. Save the Material with default values.
4. Close the Asset Editor.
5. Open the Asset Editor.
6. Select _File_ -> _Open Recent_ -> previously saved Blast Material.
7. Close the Asset Editor.
8. On the _Save Changes_ prompt select _Yes_.

**Expected behavior**
The Blast Material can be successfully loaded in the Asset Editor. The Editor remains stable.

**Video**

https://user-images.githubusercontent.com/86953108/133589737-e8e5376c-1da3-4739-b234-61324316fd50.mp4

**Call Stack Trace:**
```
25) 00007FFAA726C247 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FFAA95D5270 (ntdll) : memset
23) 00007FFAA95BC776 (ntdll) : _C_specific_handler
22) 00007FFAA95D217F (ntdll) : _chkstk
21) 00007FFAA9581454 (ntdll) : RtlRaiseException
20) 00007FFAA95D0CAE (ntdll) : KiUserExceptionDispatcher
19) 00007FFA32092CDA (EditorLib) : ModuleInitISystem
18) 00007FFA3209501C (EditorLib) : ModuleInitISystem
17) 00007FFA3160DDEA (EditorLib) : CEditorCommandManager::UnregisterCommand
16) 00007FFA508A7FA2 (Qt5Widgets) : QWidget::event
15) 00007FFA508849B0 (Qt5Widgets) : QApplicationPrivate::notify_helper
14) 00007FFA50883A33 (Qt5Widgets) : QApplication::notify
13) 00007FFA50471B4A (Qt5Core) : QCoreApplication::notifyInternal2
12) 00007FFA4FB71D12 (AzQtComponents) : AzQtComponents::StyledDockWidget::closeEvent
11) 00007FFA508A7FA2 (Qt5Widgets) : QWidget::event
10) 00007FFA50977466 (Qt5Widgets) : QDockWidget::event
 9) 00007FFA508849B0 (Qt5Widgets) : QApplicationPrivate::notify_helper
 8) 00007FFA50883A33 (Qt5Widgets) : QApplication::notify
 7) 00007FFA50471B4A (Qt5Core) : QCoreApplication::notifyInternal2
 6) 00007FFA508A49DF (Qt5Widgets) : QWidgetPrivate::close_helper
 5) 00007FFA50490DB9 (Qt5Core) : QObject::qt_static_metacall
 4) 00007FFA4FB96AB6 (AzQtComponents) : AzQtComponents::DockBarButton::handleButtonClick
 3) 00007FFA50490DB9 (Qt5Core) : QObject::qt_static_metacall
 2) 00007FFA5094FE57 (Qt5Widgets) : QAbstractButton::clicked
 1) 00007FFA5094FD78 (Qt5Widgets) : QAbstractButton::click

```

**Desktop/Device:**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU AMD Ryzen 5 3600
 - GPU Nvidia RTX 2060 SUPER
 - Memory 16GB

**Attachments**
[Editor.log](https://github.com/o3de/o3de/files/7176625/Editor.log)
[error.log](https://github.com/o3de/o3de/files/7176626/error.log)
[error.zip](https://github.com/o3de/o3de/files/7176627/error.zip)

**Comment by moraaar** [comment]
I was able to reproduce this bug using a "Pre-built SDK Engine" and I found the root cause of the issue, which related to the process of creating the "Pre-built SDK Engine".

Blast gem has a settings registry file (`Gems\Blast\AssetProcessorGemConfig.setreg`) which tells the Asset Processor how to process `.blast`, `.blastmaterial` and `.blastconfiguration` asset types. `Gems\Blast\AssetProcessorGemConfig.setreg` is not present in the "Pre-built SDK Engine", there might be some step missing to copy these .setreg files from the gem. 

Because of this, when editor is opened, Asset Processor will not copy those blast asset types to the cache, causing the crash when Asset Editor tries to open them.

Removing physics feature label, increasing priority to blocker since this could be a big issue if .setreg files from gems (or even other places) are not considered when making Pre-built SDK Engine (final product).

**Comment by moraaar** [comment]
FYI @lumberyard-employee-dm @amznestebanpapp

**Comment by amznestebanpapp** [comment]
we dont automatically copy that file, only files that are in the `Assets` folder, add this to `Gems\Blast\CMakeLists.txt`
```
ly_install_files(FILES AssetProcessorGemConfig.setreg
    DESTINATION Gems\Blast
)
```
But not sure how that file is being picked by the AP, shouldnt that be in a user or Registry folder?
If the AP is picking these files maybe we need to automatically copy them @lumberyard-employee-dm

**Comment by lumberyard-employee-dm** [comment]
The AssetProcessorGemConfig.setreg should be moved to the `Gems/Blast/Registry` folder.

The AP will pick up files in that Registry directory and we automatically pull the [<gem-root>/Registry](https://github.com/o3de/o3de/blob/development/cmake/Platform/Common/Install_common.cmake#L683-L686) folder into the install SDK layout

**Comment by moraaar** [comment]
Thanks guys, I'll move it to `Gems/Blast/Registry` folder and test it.

**Comment by LB-TomaszSerwanski** [comment]
Issue verified as fixed in Development (f5fcab7).
